test/common/http/codec_impl_fuzz_test.cc

```diff
@@ -397,6 +397,10 @@ class ReorderBuffer {
     while (!bufs_.empty()) {
       Buffer::OwnedImpl& buf = bufs_.front();
       while (buf.length() > 0) {
+        if (!can_dispatch_) {
+          ENVOY_LOG_MISC(trace, "Buffer dispatch disabled, stopping drain");
+          return codecClientError("preventing buffer drain due to connection reset");
```

Comment:
I'm going to trust @asraa for if returning a codecClientError is the semantically correct thing to do if the fuzzer is asked to violate the codec invariants.  Otherwise this LGTM, thanks!